UI/TechTreeWnd.cpp

```diff
@@ -238,139 +255,155 @@ class TechTreeWnd::TechTreeControls : public CUIWnd {
     // calculated by SizeMove, and stored, so that start and end positions don't need to be recalculated each
     // time Render is called.
 
-    std::vector<CUIButton*>             m_category_buttons;
-    std::map<TechStatus, CUIButton*>    m_tech_status_buttons;
-    CUIButton*                          m_list_view_button;
-    CUIButton*                          m_tree_view_button;
+    boost::shared_ptr<GG::SubTexture>            GetManagedSubTexture(const std::string& fn);  ///< return a shared_ptr of a managed subtexture for tech controls
+    CUIToggleButton*                             m_view_type_button;
+    CUIToggleRepresenter*                        m_view_type_representer;
+    CUIToggleButton*                             m_all_cat_button;
+    CUIToggleRepresenter*                        m_all_cat_representer;
+    std::map<std::string, CUIToggleButton*>      m_cat_buttons;
+    std::map<std::string, CUIToggleRepresenter*> m_cat_representers;
+    std::map<TechStatus, CUIToggleButton*>       m_status_buttons;
+    std::map<TechStatus, CUIToggleRepresenter*>  m_status_representers;
 
     friend class TechTreeWnd;               // so TechTreeWnd can access buttons
 };
-const int TechTreeWnd::TechTreeControls::BUTTON_SEPARATION = 3;
+
+const int TechTreeWnd::TechTreeControls::BUTTON_SEPARATION = 2;
 const int TechTreeWnd::TechTreeControls::UPPER_LEFT_PAD = 2;
 
 TechTreeWnd::TechTreeControls::TechTreeControls(const std::string& config_name) :
-    CUIWnd(UserString("TECH_DISPLAY"),
-           GG::INTERACTIVE | GG::DRAGABLE | GG::RESIZABLE | GG::ONTOP,
-           config_name)
+    CUIWnd(UserString("TECH_DISPLAY"), GG::INTERACTIVE | GG::DRAGABLE | GG::RESIZABLE | GG::ONTOP, config_name)
 {
     // create a button for each tech category...
     const std::vector<std::string>& cats = GetTechManager().CategoryNames();
     for (unsigned int i = 0; i < cats.size(); ++i) {
-        GG::Clr background_clr = ClientUI::CategoryColor(cats[i]);
-        GG::Clr border_clr     = background_clr;
-        AdjustBrightness(background_clr, -50);
-        m_category_buttons.push_back(new CUIButton(UserString(cats[i]), background_clr, border_clr));
-        AttachChild(m_category_buttons.back());
+        GG::Clr icon_clr = ClientUI::CategoryColor(cats[i]);
+        boost::shared_ptr<GG::SubTexture> icon = boost::make_shared<GG::SubTexture>(ClientUI::CategoryIcon(cats[i]));
+        m_cat_representers[cats[i]] = new CUIToggleRepresenter(icon, icon_clr);
+        m_cat_buttons[cats[i]] = new CUIToggleButton(UserString(cats[i]),
+                                                     boost::shared_ptr<CUIToggleRepresenter>(m_cat_representers[cats[i]]));
```

Comment:
Instead of `boost::shared_ptr<Type>(new Type(arg1, arg2))` you can use `boost::make_shared<Type>(arg1, arg2)` to reduce code duplication.